webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -801,91 +825,92 @@ public static float fitText(Font font, String text, Rectangle rect, float maxFon
         return potentialFontSize;
     }
 
-    private static LinkedList<Chunk> splitScrambleToLineChunks(String paddedScramble, Font scrambleFont, float scrambleColumnWidth) {
+    private static LinkedList<Chunk> splitTextToLineChunks(String text, Font scrambleFont, float scrambleColumnWidth) {
         float availableScrambleWidth = scrambleColumnWidth - 2*SCRAMBLE_PADDING_HORIZONTAL;
 
-        int startIndex = 0;
-        int endIndex = 0;
         LinkedList<Chunk> lineChunks = new LinkedList<Chunk>();
-        while(startIndex < paddedScramble.length()) {
-            // Walk forwards until we've grabbed the maximum number of characters
-            // that fit in a line, we've run out of characters, or we hit a newline.
-            float substringWidth;
-            for(endIndex++; endIndex <= paddedScramble.length(); endIndex++) {
-                if(paddedScramble.charAt(endIndex - 1) == '\n') {
-                    break;
-                }
-                String scrambleSubstring = NON_BREAKING_SPACE + paddedScramble.substring(startIndex, endIndex) + NON_BREAKING_SPACE;
-                substringWidth = scrambleFont.getBaseFont().getWidthPoint(scrambleSubstring, scrambleFont.getSize());
-                if(substringWidth > availableScrambleWidth) {
-                    break;
-                }
-            }
-            // endIndex is one past the best fit, so remove one character and it should fit!
-            endIndex--;
-
-            // If we're not at the end of the scramble, make sure we're not cutting
-            // a turn in half by walking backwards until we're right before a turn.
-            // Any spaces added for padding after a turn are considered part of
-            // that turn because they're actually NON_BREAKING_SPACE, not a ' '.
-            int perfectFitEndIndex = endIndex;
-            if(endIndex < paddedScramble.length()) {
-                while(true) {
-                    if(endIndex < startIndex) {
-                        // We walked all the way to the beginning of the line
-                        // without finding a good breaking point. Give up and break
-                        // in the middle of a turn =(.
-                        endIndex = perfectFitEndIndex;
+        String[] lineList = text.split("\n");
+        
+        for (String line:lineList){
+            int startIndex = 0;
+            int endIndex = 0;
+            while(startIndex < line.length()) {
+                // Walk forwards until we've grabbed the maximum number of characters
+                // that fit in a line, we've run out of characters, or we hit a newline.
```

Comment:
This comment says "... we hit a newline", but I don't see the code checking anywhere for a newline, it just seems to be checking if we've used up all the characters or if we've used up all the horizontal space.

I see that you added a split above that separates on newlines. Would it be cleaner to add logic here to check for newline instead? I feel like that would reduce the amount of indentation going on in this method. Alternatively, if that doesn't work, can we fix this comment so it's accurate?

Reply:
Spliting into lines makes this function fit for single and multi line text, so we can use it in rules, megaminx scrambles and single lines text. I've fixed the comments and kept the logic :smile: